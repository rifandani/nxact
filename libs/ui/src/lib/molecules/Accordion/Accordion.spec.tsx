import renderer from 'react-test-renderer';
import {
  Accordion,
  AccordionContent,
  AccordionItem,
  AccordionTrigger,
} from './Accordion';

describe('Collapsible component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Accordion type="single" collapsible>
          <AccordionItem value="item-1">
            <AccordionTrigger>Is it accessible?</AccordionTrigger>
            <AccordionContent>
              Yes. It adheres to the WAI-ARIA design pattern.
            </AccordionContent>
          </AccordionItem>
          <AccordionItem value="item-2">
            <AccordionTrigger>Is it styled?</AccordionTrigger>
            <AccordionContent>
              Yes. It comes with default styles that matches the other
              components' aesthetic.
            </AccordionContent>
          </AccordionItem>
          <AccordionItem value="item-3">
            <AccordionTrigger>Is it animated?</AccordionTrigger>
            <AccordionContent>
              Yes. It's animated by default, but you can disable it if you
              prefer.
            </AccordionContent>
          </AccordionItem>
        </Accordion>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
