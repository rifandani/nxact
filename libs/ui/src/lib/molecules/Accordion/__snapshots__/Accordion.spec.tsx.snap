// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapsible component should render correctly 1`] = `
<div
  data-orientation="vertical"
  onKeyDown={[Function]}
>
  <div
    className="border-b border-b-slate-200 dark:border-b-slate-700"
    data-orientation="vertical"
    data-state="closed"
  >
    <h3
      className="flex"
      data-orientation="vertical"
      data-state="closed"
    >
      <button
        aria-controls="radix-:r1:"
        aria-expanded={false}
        className="flex flex-1 items-center justify-between py-4 font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180"
        data-orientation="vertical"
        data-radix-collection-item=""
        data-state="closed"
        id="radix-:r0:"
        onClick={[Function]}
        type="button"
      >
        Is it accessible?
        <svg
          className="h-4 w-4 rotate-90 transition-transform duration-200"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeWidth={2}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.5 18.5L15.5 11.5L8.5 4.5"
          />
        </svg>
      </button>
    </h3>
    <div
      aria-labelledby="radix-:r0:"
      className="data-[state=open]:animate-accordion-down data-[state=closed]:animate-accordion-up overflow-hidden text-sm transition-all"
      data-orientation="vertical"
      data-state="closed"
      hidden={true}
      id="radix-:r1:"
      role="region"
      style={
        Object {
          "--radix-accordion-content-height": "var(--radix-collapsible-content-height)",
          "--radix-accordion-content-width": "var(--radix-collapsible-content-width)",
          "--radix-collapsible-content-height": undefined,
          "--radix-collapsible-content-width": undefined,
        }
      }
    />
  </div>
  <div
    className="border-b border-b-slate-200 dark:border-b-slate-700"
    data-orientation="vertical"
    data-state="closed"
  >
    <h3
      className="flex"
      data-orientation="vertical"
      data-state="closed"
    >
      <button
        aria-controls="radix-:r3:"
        aria-expanded={false}
        className="flex flex-1 items-center justify-between py-4 font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180"
        data-orientation="vertical"
        data-radix-collection-item=""
        data-state="closed"
        id="radix-:r2:"
        onClick={[Function]}
        type="button"
      >
        Is it styled?
        <svg
          className="h-4 w-4 rotate-90 transition-transform duration-200"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeWidth={2}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.5 18.5L15.5 11.5L8.5 4.5"
          />
        </svg>
      </button>
    </h3>
    <div
      aria-labelledby="radix-:r2:"
      className="data-[state=open]:animate-accordion-down data-[state=closed]:animate-accordion-up overflow-hidden text-sm transition-all"
      data-orientation="vertical"
      data-state="closed"
      hidden={true}
      id="radix-:r3:"
      role="region"
      style={
        Object {
          "--radix-accordion-content-height": "var(--radix-collapsible-content-height)",
          "--radix-accordion-content-width": "var(--radix-collapsible-content-width)",
          "--radix-collapsible-content-height": undefined,
          "--radix-collapsible-content-width": undefined,
        }
      }
    />
  </div>
  <div
    className="border-b border-b-slate-200 dark:border-b-slate-700"
    data-orientation="vertical"
    data-state="closed"
  >
    <h3
      className="flex"
      data-orientation="vertical"
      data-state="closed"
    >
      <button
        aria-controls="radix-:r5:"
        aria-expanded={false}
        className="flex flex-1 items-center justify-between py-4 font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180"
        data-orientation="vertical"
        data-radix-collection-item=""
        data-state="closed"
        id="radix-:r4:"
        onClick={[Function]}
        type="button"
      >
        Is it animated?
        <svg
          className="h-4 w-4 rotate-90 transition-transform duration-200"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeWidth={2}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.5 18.5L15.5 11.5L8.5 4.5"
          />
        </svg>
      </button>
    </h3>
    <div
      aria-labelledby="radix-:r4:"
      className="data-[state=open]:animate-accordion-down data-[state=closed]:animate-accordion-up overflow-hidden text-sm transition-all"
      data-orientation="vertical"
      data-state="closed"
      hidden={true}
      id="radix-:r5:"
      role="region"
      style={
        Object {
          "--radix-accordion-content-height": "var(--radix-collapsible-content-height)",
          "--radix-accordion-content-width": "var(--radix-collapsible-content-width)",
          "--radix-collapsible-content-height": undefined,
          "--radix-collapsible-content-width": undefined,
        }
      }
    />
  </div>
</div>
`;
